core/common/src/main/java/alluxio/security/authentication/SaslStreamClientDriver.java

```diff
@@ -111,8 +111,26 @@ public void onNext(SaslMessage saslMessage) {
 
   @Override
   public void onError(Throwable throwable) {
-    LOG.warn("Received error on client driver for channel: {}. Error: {}", mChannelId, throwable);
     mHandshakeFuture.setException(throwable);
+
+    // Whether to trace the error as debug.
+    boolean traceDebug = false;
+    // UNAVAILABLE is quite common while cluster is booting up.
+    // Tracing it as warning will pollute logs as onError() will be called per retry.
+    if (throwable instanceof StatusRuntimeException) {
+      if (((StatusRuntimeException) throwable).getStatus().getCode() == Status.UNAVAILABLE
+          .getCode()) {
+        traceDebug = true;
+      }
+    }
+
+    String errorMsg = String.format("Received error on client driver for channel: %s. Error: %s",
+        mChannelId, throwable);
+    if (traceDebug) {
+      LOG.debug(errorMsg);
+    } else {
+      LOG.warn(errorMsg);
+    }
```

Comment:
Could we simplify this block of code by doing the conditional check after we already created the error message?

Reply:
Done